### PR TITLE
User Suggestion, Search Bar, User Routing

### LINKED ISSUE #86: Few Additional componens
- [x] User Suggestion Component
- [x] Search User

### LINKED ISSUE #87: Add Routing to User Profile as user/id
- View profile of users by navigating to `/user/user_id`

### PR DESCRIPTION
**Pull request by bodhisha**
### Resolves #86 
**Changes**

- User suggestion UI and Integeration
- Search bar Integeration
- Add User Routing to user/id to view each user profile

### CODE REVIEW

File: frontend/src/Components/Suggestions.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                xmlns="http://www.w3.org/2000/svg"
+                width="10"
+                fill="currentColor"
+                class="bi bi-image-fill"
+                viewBox="0 0 16 16"
+              >
+                <path d="M.002 3a2 2 0 0 1 2-2h12a2 2 0 0 1 2 2v10a2 2 0 0 1-2 2h-12a2 2 0 0 1-2-2V3zm1 9v1a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1V9.5l-3.777-1.947a.5.5 0 0 0-.577.093l-3.71 3.71-2.66-1.772a.5.5 0 0 0-.63.062L1.002 12zm5-6.5a1.5 1.5 0 1 0-3 0 1.5 1.5 0 0 0 3 0z" />
+              </svg>
+            </span>
+            5 Posts
```
Comment (aitikgupta) [inline frontend/src/Components/Suggestions.js]: Can you add one more commit which can return the length of the posts in the lightweight call? (in the backend? - Let's do this right)

File: frontend/src/Components/UserProfile.js [hunk 2]
```diff
@@ -1,26 +1,24 @@
...
-      })
+      .get(GET_USER_DETAILS + `${id}`)
       .then((res) => {
-        setPosts(res.data.data);
         setLoading(false);
+        setUser(res.data.data);
+      })
+      .catch((err) => {
+        setLoading(false);
+        console.log("User Not found"); //catch error in backend and update here
```
Comment (aitikgupta) [inline frontend/src/Components/UserProfile.js]: We can simply `toast.error("User does not exist")`?

**Review by DebugAgrawal — APPROVED** [review]

File: frontend/src/constants.js [hunk 3]
```diff
@@ -17,6 +17,8 @@ export const POST_CREATE_URL = `${BASE_URL}/post/create`;
 export const POST_GET_ALL_URL = `${BASE_URL}/post/all`;
 export const POST_COMMENT_URL = `${BASE_URL}/post/comment`;
 export const POST_UNCOMMENT_URL = `${BASE_URL}/post/uncomment`;
+export const USER_SUGGESTIONS = `${BASE_URL}/suggestions`;
+export const GET_USER_DETAILS = `${BASE_URL}/user/details?user_id=`;
```
Comment (aitikgupta) [inline frontend/src/constants.js]: Can we change the variable name to have a `_URL` at the end? (for consistency)
`USER_SUGGESTIONS_LIGHT_URL` or `GET_USER_DETAILS_URL`?

**Review by aitikgupta — APPROVED** [review]
Leaving a few comments, I think this is amazing! 🎉

File: frontend/src/Components/Suggestions.js (hunk 1 above)
Comment (bodhisha) [reply frontend/src/Components/Suggestions.js]: As discussed, removing this div

File: frontend/src/Components/UserProfile.js (hunk 2 above)
Comment (bodhisha) [reply frontend/src/Components/UserProfile.js]: Done

File: frontend/src/constants.js (hunk 3 above)
Comment (bodhisha) [reply frontend/src/constants.js]: Done

**Pull request merged**